Crash: Null Reference Exception

**Root Cause**
The DayTemplate is being initialized last which causes a race condition and a crash when attempting to draw the screen.

**Workaround**
If every property is Binded to a ViewModel the race condition does not happen

**Stacktrace**
```
System.NullReferenceException: Object reference not set to an instance of an object.

05-21 16:23:52.229 D/Mono    (19927): DllImport attempting to load: '/system/lib/liblog.so'.
05-21 16:23:52.229 D/Mono    (19927): DllImport loaded library '/system/lib/liblog.so'.
05-21 16:23:52.229 D/Mono    (19927): DllImport searching in: '/system/lib/liblog.so' ('/system/lib/liblog.so').
05-21 16:23:52.229 D/Mono    (19927): Searching for '__android_log_print'.
05-21 16:23:52.229 D/Mono    (19927): Probing '__android_log_print'.
05-21 16:23:52.229 D/Mono    (19927): Found as '__android_log_print'.
05-21 16:23:52.244 I/MonoDroid(19927): UNHANDLED EXCEPTION:
05-21 16:23:52.348 I/MonoDroid(19927): System.NullReferenceException: Object reference not set to an instance of an object.
05-21 16:23:52.348 I/MonoDroid(19927):   at XamCal.Abstractions.Calendar.CreateView (XamCal.Abstractions.Day day) [0x00000] in C:\projects\xamcal\src\XamCal.Abstractions\Calendar.cs:65 
05-21 16:23:52.348 I/MonoDroid(19927):   at XamCal.Abstractions.Calendar.Redraw () [0x0005a] in C:\projects\xamcal\src\XamCal.Abstractions\Calendar.cs:51 
05-21 16:23:52.348 I/MonoDroid(19927):   at XamCal.Abstractions.Calendar.OnMonthPropertyChanged (Xamarin.Forms.BindableObject bindable, System.Object oldValue, System.Object newValue) [0x00016] in C:\projects\xamcal\src\XamCal.Abstractions\Calendar.BindableProperties.cs:55 
05-21 16:23:52.348 I/MonoDroid(19927):   at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) [0x0011b] in D:\agent\_work\1\s\Xamarin.Forms.Core\BindableObject.cs:597 
05-21 16:23:52.348 I/MonoDroid(19927):   at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x0015b] in D:\agent\_work\1\s\Xamarin.Forms.Core\BindableObject.cs:391 
05-21 16:23:52.348 I/MonoDroid(19927):   at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindableProperty property, System.Object value, System.Boolean fromStyle, System.Boolean checkAccess) [0x0005f] in D:\agent\_work\1\s\Xamarin.Forms.Core\BindableObject.cs:544 
05-21 16:23:52.348 I/MonoDroid(19927):   at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindableProperty property, System.Object value) [0x00000] in D:\agent\_work\1\s\Xamarin.Forms.Core\BindableObject.cs:84 
05-21 16:23:52.348 I/MonoDroid(19927):   at SampleCal.MainPage.InitializeComponent () [0x00012] in D:\XamCal\sample\SampleCal\obj\Debug
etstandard2.0\SampleCal.MainPage.xaml.g.cs:20 
05-21 16:23:52.348 I/MonoDroid(19927):   at SampleCal.MainPage..ctor () [0x00008] in D:\XamCal\sample\SampleCal\MainPage.xaml.cs:14 
05-21 16:23:52.348 I/MonoDroid(19927):   at SampleCal.App..ctor () [0x0000f] in D:\XamCal\sample\SampleCal\App.xaml.cs:14 
05-21 16:23:52.348 I/MonoDroid(19927):   at SampleCal.Droid.MainActivity.OnCreate (Android.OS.Bundle bundle) [0x0002d] in D:\XamCal\sample\SampleCal.Android\MainActivity.cs:25 
05-21 16:23:52.348 I/MonoDroid(19927):   at Android.App.Activity.n_OnCreate_Landroid_os_Bundle_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_savedInstanceState) [0x0000f] in <25661073a35344a89f215a4cf81af37c>:0 
05-21 16:23:52.348 I/MonoDroid(19927):   at (wrapper dynamic-method) System.Object.f4de637a-3ab9-49ff-8be9-8bbd1bec1d73(intptr,intptr,intptr)
05-21 16:23:52.375 W/art     (19927): JNI RegisterNativeMethods: attempt to register 0 native methods for android.runtime.JavaProxyThrowable
05-21 16:23:52.382 D/Mono    (19927): DllImport searching in: '__Internal' ('(null)').
05-21 16:23:52.382 D/Mono    (19927): Searching for 'java_interop_jnienv_throw'.
05-21 16:23:52.382 D/Mono    (19927): Probing 'java_interop_jnienv_throw'.
05-21 16:23:52.383 D/Mono    (19927): Found as 'java_interop_jnienv_throw'.
An unhandled exception occured.

05-21 16:23:53.481 E/mono    (19927): 
05-21 16:23:53.481 E/mono    (19927): Unhandled Exception:
05-21 16:23:53.481 E/mono    (19927): System.NullReferenceException: Object reference not set to an instance of an object.
05-21 16:23:53.481 E/mono    (19927):   at (wrapper dynamic-method) System.Object.f4de637a-3ab9-49ff-8be9-8bbd1bec1d73(intptr,intptr,intptr)
05-21 16:23:53.481 E/mono-rt (19927): [ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object.
05-21 16:23:53.481 E/mono-rt (19927):   at (wrapper dynamic-method) System.Object.f4de637a-3ab9-49ff-8be9-8bbd1bec1d73(intptr,intptr,intptr)
```

Found while working on #1

--- a/src/XamCal.Abstractions/Calendar.cs
+++ b/src/XamCal.Abstractions/Calendar.cs
@@ -38,7 +38,7 @@ namespace XamCal.Abstractions
 
         private void Redraw()
         {
-            if (Year <= 0 || Month <= 0) return;
+            if (Year <= 0 || Month <= 0 || DayTemplate == null) return;
 
             var daysInMonth = DateTime.DaysInMonth(Year, Month);
             _grid.Children.Clear();

